src/v/cluster/partition_leaders_table.h

```diff
@@ -98,6 +98,21 @@ class partition_leaders_table {
 
     void reset() { _leaders.clear(); }
 
+    struct leader_info_t {
```

Comment:
Another wait do `leader_key...` and `leader_meta` public and return ref to hash_map. But I decided do not move this stuff to public section